### PR TITLE
fix(core/textarea): textareaRows, textareaWidth, and textareaHeight

### PR DESCRIPTION
**Pull request by nuke-ellington**
## 💡 What is the current behavior?

- Unit conversion issues: The `textareaWidth` and `textareaHeight` properties only accepted raw values without proper unit conversion support for `rem`, `em`, `%`, or unitless values.

- CSS height conflicts: When using the `rows` attribute, there was a conflict with the fixed `height: 3.25rem` CSS rule, preventing the browser's natural height calculation based on row count.

GitHub Issue Number: #1975

## 🆕 What is the new behavior?

- Enhanced unit conversion: Added comprehensive `convertToPx()` method that handles `px`, `rem`, `em`, `%`, and unitless values with proper error handling and case-insensitive unit detection.

- Improved CSS styling: Modified textarea mixin to conditionally apply fixed height only when `rows` attribute is not present `[&:not([rows])`, allowing natural browser sizing when rows are specified.

- Robust dimension handling: Enhanced resize behavior management with proper fallbacks (`'auto'` for `height`, `'100%'` for `width`) and improved manual resize state preservation.

- Comprehensive test coverage: Added extensive component tests covering unit conversion, dimension interactions, resize behavior, and edge cases.

## 🏁 Checklist

A pull request can only be merged if all of these conditions are met (where applicable):

- [ ] 🦮 Accessibility (a11y) features were implemented
- [ ] 🗺️ Internationalization (i18n) - no hard coded strings
- [ ] 📲 Responsiveness - components handle viewport changes and content overflow gracefully
- [ ] 📕 Add or update a Storybook story
- [x] 📄 Documentation was reviewed/updated [siemens/ix-docs](https://github.com/siemens/ix-docs)
- [x] 🧪 Unit tests were added/updated and pass (`pnpm test`)
- [x] 📸 Visual regression tests were added/updated and pass ([Guide](https://github.com/siemens/ix/blob/main/CONTRIBUTING.md#visual-regression-testing))
- [x] 🧐 Static code analysis passes (`pnpm lint`)
- [x] 🏗️ Successful compilation (`pnpm build`, changes pushed)

### CODE REVIEW

File: packages/core/src/components/input/textarea.tsx [hunk 1]
```diff
@@ -292,6 +333,51 @@ export class Textarea implements IxInputFieldComponent<string> {
...
+    const convertedHeight = this.getConvertedDimension(
+      this.textareaHeight,
+      'height'
+    );
+
+    if (convertedHeight) {
+      return convertedHeight;
+    }
+
+    return this.textareaRows ? 'auto' : undefined;
```
Comment (alexkaduk) [inline packages/core/src/components/input/textarea.tsx]: The rows behavior could be handled in CSS so the component doesn’t need to send height: 'auto'.
In packages/core/scss/mixins/_input.scss, in the element-textarea mixin, apply the fixed height only when rows is not set:
```
&:not([rows]) {
  height: 3.25rem;
}
```
After
`return this.textareaRows ? 'auto' : undefined;` can be just `return undefined;`

File: packages/core/src/components/input/textarea.tsx [hunk 2]
```diff
@@ -292,6 +333,51 @@ export class Textarea implements IxInputFieldComponent<string> {
...
+    const convertedWidth = this.getConvertedDimension(
+      this.textareaWidth,
+      'width'
+    );
+
+    if (convertedWidth) {
+      return convertedWidth;
+    }
+
+    return this.textareaCols ? 'auto' : undefined;
```
Comment (alexkaduk) [inline packages/core/src/components/input/textarea.tsx]: same https://github.com/siemens/ix/pull/2293/changes#r2933983818 here.

File: packages/core/src/components/input/textarea.tsx [hunk 3]
```diff
@@ -292,6 +333,51 @@ export class Textarea implements IxInputFieldComponent<string> {
...
+    dimension: 'width' | 'height'
+  ): string | undefined {
+    return convertToPx(
+      value,
+      dimension,
+      this.textAreaRef.current || this.hostElement
+    );
+  }
+
+  private getTextareaHeight(): string | undefined {
```
Comment (alexkaduk) [inline packages/core/src/components/input/textarea.tsx:336]: it is very similar to `getTextareaWidth`. pls consider smth like this
```
private getResolvedDimension(
  kind: 'height' | 'width',
  manual: string | undefined,
  fromProp: string | undefined,
  useAuto: boolean,
  fallback?: string
): string | undefined {
  if (this.isManuallyResized && manual) return manual;
  const value = normalizeDimension(fromProp);
  if (value) return value;
  if (useAuto) return 'auto';
  return fallback;
}
```

File: packages/core/src/components/utils/unit-conversion.util.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+ * Normalizes CSS unit values used for dimension props.
+ *
+ * - Explicit CSS units (`px`, `rem`, `em`, `%`) are preserved.
+ * - Unitless numeric values are converted to pixels.
+ *
+ * @param value - The CSS value to normalize (e.g., "16px", "1rem", "100%")
+ * @param dimension - The target dimension (`width` or `height`)
+ * @returns A normalized CSS value or undefined if parsing fails
+ */
+export function convertToPx(
```
Comment (alexkaduk) [inline packages/core/src/components/utils/unit-conversion.util.ts]: The name convertToPx suggests everything becomes pixels. At the same time, only unitless values get px; rem/em/% are passed through. The name might be:
`normalizeDimension` or `parseDimension`

Comment (alexkaduk) [reply packages/core/src/components/utils/unit-conversion.util.ts]: Consider to remove unused _dimension and _element.

Comment (alexkaduk) [reply packages/core/src/components/utils/unit-conversion.util.ts]: Might make sense to add unit tests. It allows to drop or shrink packages/core/src/components/input/tests/textarea-dimensions.ct.ts

Comment (alexkaduk) [reply packages/core/src/components/utils/unit-conversion.util.ts]: Optional: reject negative numbers.

File: packages/core/src/components/input/textarea.tsx (hunk 3 above)
Comment (nuke-ellington) [reply packages/core/src/components/input/textarea.tsx:336]: Tried that but it makes the code hard to read IMO. Refactored the functions to call `normalizeCssDimension` for less redundancy.
